core/src/main/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworks.java

```diff
@@ -56,155 +68,118 @@ public PrepareRoutingSubnetworks(GraphHopperStorage ghStorage, List<PrepareJob>
-        return removedEdgeKeys;
-    }
-
-    private int setSubnetwork(int edgeKey, EdgeFilter filter, BooleanEncodedValue inSubnetworkEnc) {
-        int edgeId = EdgeBasedTarjanSCC.getEdgeFromKey(edgeKey);
-        EdgeIteratorState edge = ghStorage.getEdgeIteratorState(edgeId, Integer.MIN_VALUE);
-        if (edgeKey % 2 == 0 && filter.accept(edge)) {
-            edge.set(inSubnetworkEnc, true);
-            return 1;
-        }
-        if (edgeKey % 2 != 0 && filter.accept(edge.detach(true))) {
-            edge.set(inSubnetworkEnc, true);
-            return 1;
-        }
-        return 0;
+        logger.info(subnetworkEnc.getName().replaceAll("_subnetwork", "") + " - Marked " + subnetworks + " subnetworks (biggest: " + biggestSubnetwork + " edges) -> " +
+                (ccs.getTotalComponents() - subnetworks) + " components(s) remain (smallest: " + smallestNonSubnetwork + ", biggest: " + ccs.getBiggestComponent().size() + " edges)"
+                + ", total marked edges: " + markedEdges + ", took: " + sw.stop().getSeconds() + "s");
+        return markedEdges;
     }
 
-    /**
-     * This method checks if the node is removed or inaccessible for ALL encoders.
-     *
-     * @return true if no edges are reachable from the specified nodeIndex for any flag encoder.
-     */
-    boolean detectNodeRemovedForAllEncoders(EdgeExplorer edgeExplorerAllEdges, int nodeIndex) {
-        // we could implement a 'fast check' for several previously marked removed nodes via GHBitSet 
-        // removedNodesPerVehicle. The problem is that we would need long-indices but BitSet only supports int (due to nodeIndex*numberOfEncoders)
+    private int setSubnetworkEdge(int edgeKey, Weighting weighting, BooleanEncodedValue subnetworkEnc) {
+        // edges that are not accessible anyway are not marked as subnetworks additionally
+        if (!Double.isFinite(weighting.calcEdgeWeightWithAccess(ghStorage.getEdgeIteratorStateForKey(edgeKey), false)))
+            return 0;
 
-        List<EdgeFilter> inSubnetworkEncs = new ArrayList<>();
-        for (PrepareJob job : prepareJobs) {
-            inSubnetworkEncs.add(createEdgeFilter(job.weighting, job.inSubnetworkEnc));
-        }
-        // if no edges are reachable return true
-        EdgeIterator iter = edgeExplorerAllEdges.setBaseNode(nodeIndex);
-        while (iter.next()) {
-            // if at least one encoder allows one direction return false
-            for (EdgeFilter edgeFilter : inSubnetworkEncs) {
-                if (edgeFilter.accept(iter))
-                    return false;
-            }
+        int edge = EdgeBasedTarjanSCC.getEdgeFromKey(edgeKey);
+        EdgeIteratorState edgeState = ghStorage.getEdgeIteratorState(edge, Integer.MIN_VALUE);
+        if (!edgeState.get(subnetworkEnc)) {
+            edgeState.set(subnetworkEnc, true);
+            return 1;
+        } else {
+            return 0;
```

Comment:
Previously we used the EdgeFilter from createEdgeFilter here, but I think this way the code is more clear.